Readability.js

```diff
@@ -245,10 +260,24 @@ Readability.prototype = {
       if (curTitle.split(' ').length < 3)
         curTitle = origTitle.replace(/[^\|\-]*[\|\-](.*)/gi,'$1');
     } else if (curTitle.indexOf(': ') !== -1) {
-      curTitle = origTitle.replace(/.*:(.*)/gi, '$1');
+      // XXX check if we have an heading containing this exact string, so we
+      // could assume it's the full title
+      // var headings = [].concat.call([].slice.call(doc.getElementsByTagName('h1')),
+      //                               [].slice.call(doc.getElementsByTagName('h2')));
+      var headings = this._concatNodeLists(
+        doc.getElementsByTagName('h1'),
+        doc.getElementsByTagName('h2')
+      );
+      var match = this._someNode(headings, function(heading) {
+        return heading.textContent === curTitle;
+      });
 
-      if (curTitle.split(' ').length < 3)
-        curTitle = origTitle.replace(/[^:]*[:](.*)/gi,'$1');
+      if (!match) {
+        curTitle = origTitle.replace(/.*:(.*)/gi, '$1');
+
+        if (curTitle.split(' ').length < 3)
+          curTitle = origTitle.replace(/[^:]*[:](.*)/gi,'$1');
```

Comment:
I know that, but you added another replace before it... so now there are two `replace` calls that strip out "foo:" and the second one overrides the first, but it does almost the same thing because it uses the same source string (instead of being cumulative).